x/ccv/provider/keeper/proposal.go

```diff
@@ -268,23 +288,23 @@ func (k Keeper) IteratePendingCreateProposal(ctx sdk.Context) {
 			panic(fmt.Errorf("failed to parse pending client key: %w", err))
 		}
 
-		var clientInfo types.CreateConsumerChainProposal
-		k.cdc.MustUnmarshal(iterator.Value(), &clientInfo)
+		var prop types.CreateConsumerChainProposal
+		k.cdc.MustUnmarshal(iterator.Value(), &prop)
+		// Populate info that's only stored in key, not store value
+		prop.ChainId = chainID
+		prop.SpawnTime = spawnTime
 
 		if !ctx.BlockTime().Before(spawnTime) {
-			err := k.CreateConsumerClient(ctx, chainID, clientInfo.InitialHeight, clientInfo.LockUnbondingOnTimeout)
-			if err != nil {
-				panic(fmt.Errorf("consumer client could not be created: %w", err))
-			}
-			execProposals = append(execProposals,
-				types.CreateConsumerChainProposal{ChainId: chainID, SpawnTime: spawnTime})
+			propsToExecute = append(propsToExecute, prop)
 		} else {
+			// No more proposals to check, since they're stored/ordered by timestamp.
 			break
 		}
 	}
 
-	// delete the proposals executed
-	k.DeletePendingCreateProposal(ctx, execProposals...)
+	// delete the proposals to be executed
+	k.DeletePendingCreateProposal(ctx, propsToExecute...)
```

Comment:
Move this out to IteratePendingCreateProposal so that everything happens in the same order. As is, your code calls DeletePendingCreateProposal before CreateConsumerClient, which is different than how it used to be.